ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiVersionService.scala

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.apiserver.services
+
+import com.daml.dec.DirectExecutionContext
+import com.daml.ledger.api.v1.version_service.GetApiVersionRequest
+import com.daml.ledger.api.v1.version_service.GetApiVersionResponse
+import com.daml.ledger.api.v1.version_service.VersionServiceGrpc
+import com.daml.ledger.api.v1.version_service.VersionServiceGrpc.VersionService
+import com.daml.platform.api.grpc.GrpcApiService
+import io.grpc.ServerServiceDefinition
+
+import scala.concurrent.Future
+import scala.io.Source
+
+private[apiserver] final class ApiVersionService private
+    extends VersionService
+    with GrpcApiService {
+
+  private val versionFile: String = "ledger-api/VERSION"
+  private lazy val apiVersion: String = readVersion(versionFile)
+
+  override def getApiVersion(request: GetApiVersionRequest): Future[GetApiVersionResponse] =
+    Future.successful {
+      GetApiVersionResponse(apiVersion)
+    }
+
+  private def readVersion(versionFileName: String): String =
+    Source
```

Comment:
Could you wrap this in a `Try` and log on `error` level in case something goes wrong?

Reply:
Do you expect exceptions to be re-thrown there or logged and swallowed?